src/rgw/rgw_coroutine.cc

```diff
@@ -309,6 +323,12 @@ RGWCoroutinesStack *RGWCoroutinesStack::spawn(RGWCoroutine *op, bool wait)
   return spawn(NULL, op, wait);
 }
 
+RGWCoroutinesStack *RGWCoroutinesStack::prealloc_stack()
+{
+    preallocated_stack = env->manager->allocate_stack();
```

Comment:
if `preallocated_stack` is already allocated, it will leak here. consider storing it in a boost::intrusive_ptr

Reply:
@cbodley I was considering to not allocate stack if preallocated stack existed, considering it a bug if prealloc_stack() is called twice without spawn. Can add that.

Reply:
@cbodley pushed this change